ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -717,6 +728,74 @@ impl fmt::Display for AddInstruction {
     }
 }
 
+/// The operand for a [Instruction::Icmp]
+///
+/// # Semantics
+///
+/// Compares two operands.
+///
+#[derive(Debug)]
+pub struct IcmpInstruction {
+    op1: PackedOperand,
+    op2: PackedOperand,
+}
+
+impl IcmpInstruction {
+    pub(crate) fn new(op1: Operand, op2: Operand) -> Self {
+        Self {
+            op1: PackedOperand::new(&op1),
+            op2: PackedOperand::new(&op2),
+        }
+    }
+
+    fn op1(&self) -> Operand {
+        self.op1.get()
+    }
+
+    fn op2(&self) -> Operand {
+        self.op2.get()
+    }
+}
+
+impl fmt::Display for IcmpInstruction {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "Icmp {}, {}", self.op1(), self.op2())
+    }
+}
+
+/// The operand for a [Instruction::Guard]
+///
+/// # Semantics
+///
+/// Guards a trace against diverging execution.
```

Comment:
I'd add some guard blurb like:

> The remainder of the trace will be compiled under the assumption that (at runtime) the guard condition is true. If the guard condition is false, then execution may not continue, and deoptimisation must occur.

Reply:
Fixed in [47682cf](https://github.com/ykjit/yk/pull/996/commits/47682cfe2dc8f9266fa615c6291ac265f348cf8c)